test/functional/pdo_sqlsrv/pdo_1027_query_timeout.phpt

```diff
@@ -0,0 +1,198 @@
+--TEST--
+GitHub issue 1027 - PDO::SQLSRV_ATTR_QUERY_TIMEOUT had no effect on PDO::exec()
+--DESCRIPTION--
+This test verifies that setting PDO::SQLSRV_ATTR_QUERY_TIMEOUT correctly should affect PDO::exec() as in the case for PDO::prepare() (as statement attribute or option).
+--ENV--
+PHPT_EXEC=true
+--SKIPIF--
+<?php require('skipif_mid-refactor.inc'); ?>
+--FILE--
+<?php
+require_once("MsSetup.inc");
+require_once("MsCommon_mid-refactor.inc");
+
+const _DELAY = 5;
+
+$message = '*Invalid value timeout specified for option PDO::SQLSRV_ATTR_QUERY_TIMEOUT.';
+$delay = _DELAY;
+$query = "WAITFOR DELAY '00:00:$delay'; SELECT 1";
+$error = '*Query timeout expired';
+
+function testTimeoutAttribute($conn, $timeout, $statementLevel = false)
+{
+    global $message;
+
+    $error = str_replace('timeout', $timeout, $message);
+
+    try {
+        if ($statementLevel) {
+            trace("statement option expects error: $error\n");
+            $options = array(PDO::SQLSRV_ATTR_QUERY_TIMEOUT => $timeout);
+            $sql = 'SELECT 1';
+            $stmt = $conn->prepare($sql, $options);
+        } else {
+            trace("connection attribute expects error: $error\n");

```

Comment:
I'm confused, do we expect a timeout error on 'SELECT 1'?

Reply:
This method set invalid timeout values so the setting should have already failed

Reply:
But it wouldn't return `Query timeout expired`, it should be `Invalid timeout value...` (so fnmatch should be comparing to $message not to $error)

Reply:
oh... here in this function there's a local variable $error as well. My bad I should have used a diff name for this local variable